pkgs/applications/blockchains/solana/default.nix

```diff
@@ -0,0 +1,90 @@
+{ lib
+, stdenv
+, clang
+, fetchFromGitHub
+, llvm
+, llvmPackages
+, openssl
+, pkg-config
+, rustPlatform
+, udev
+, zlib
+, rocksdb
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "solana";
+  # Please stick to mainnet releases here.
+  version = "1.5.19";
+
+  src = fetchFromGitHub {
+    owner = "solana-labs";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "04d2rz7hbb57w8zz0bcgkwggx2pz45pgad6imz9qsnqidwv4zc1v";
+  };
+
+  cargoSha256 = "0j6j9ivzb430pk154nnn9bm6gw9rh4frk4gvq6iv0nmlnvgqfg42";
+
+  nativeBuildInputs = [
+    pkg-config
+    llvmPackages.clang
+  ];
+
+  buildInputs = [
+    llvm
+    llvmPackages.libclang
+    openssl
+    udev
+    zlib
+    rocksdb
+  ];
+
+  LLVM_CONFIG_PATH = "${llvm}/bin/llvm-config";
+  LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
+
+  # Used by build.rs in the rocksdb-sys crate. If we don't set these, it would
+  # try to build RocksDB from source.
+  ROCKSDB_INCLUDE_DIR = "${rocksdb}/include";
+  ROCKSDB_LIB_DIR = "${rocksdb}/lib";
```

Comment:
Can we export those in preBuild?

Reply:
Done.

Note that this PR is a work in progress, the Rust binaries build now, but the `cargo-bpf` tools refer to the BPF SDK which is not yet packaged. (Upstream there’s a script that downloads prebuilt binaries from GitHub, the binaries include a patched Rust + LLVM toolchain.)